docs/introduction/how-to-work-with-money.md

```diff
@@ -0,0 +1,388 @@
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+// ...
+
+$orderItemFormBuilder->add('priceWithVat', MoneyType::class, [
+    'scale' => 6,
+    'constraints' => [
+        new NotBlank(['message' => 'Please enter unit price with VAT']),
+    ],
+]);
+```
+
+The form type is configured with a model data transformer that converts the value into a `Money` object automatically (`\Shopsys\FrameworkBundle\Form\Transformers\NumericToMoneyTransformer`).
+
+In Shopsys Framework, the default value of the `currency` option is `false` instead of `EUR`, hiding the currency symbol by default.
+
+*Note: For non-monetary numeric values use `NumberType` (see [Symfony docs](https://symfony.com/doc/3.4/reference/forms/types/number.html) for details).*
+
+### Form Constraints
+
+There are two form constraints to be used specifically with the `MoneyType` form fields.
+Because of model data transformation, you cannot use constraints that are validating scalar values (eg. `GreaterThan`).
+
+#### NotNegativeMoneyAmount
+
+Validates that the amount of money is greater or equal to zero.
+
+It has only the `message` option specifying the validation error message in case the entered value is negative.
+
+```php
+use Shopsys\FrameworkBundle\Form\Constraints\NotNegativeMoneyAmount;
+use Symfony\Component\Form\Extension\Core\Type\MoneyType;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+// ...
+
+$priceTableFormBuilder->add($key, MoneyType::class, [
+    'scale' => 6,
+    'required' => true,
+    'invalid_message' => 'Please enter price in correct format (a number with decimal separator)',
+    'constraints' => [
+        new NotBlank(['message' => 'Please enter price']),
+        new NotNegativeMoneyAmount(['message' => 'Price must be greater or equal to zero']),
+    ],
+]);
+```
+
+#### MoneyRange
+
+Similarly to [the Symfony `Range` constraint](https://symfony.com/doc/3.4/reference/constraints/Range.html), it validates that the amount of money is between some minimum and maximum.
+
+It has four options:
+- `min` specifies the minimum value, has to be an instance of `Money` or `null`
+- `max` specifies the maximum value, has to be an instance of `Money` or `null`
+- `minMessage` specifies the validation error message in case the entered value is less than the `min` value
+- `maxMessage` specifies the validation error message in case the entered value is greater than the `max` value
+
+At least one of the `min` and `max` options has to be provided for the constraint to make sense.
+
+```php
+use Shopsys\FormTypesBundle\MultidomainType;
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Form\Constraints\MoneyRange;
+use Symfony\Component\Form\Extension\Core\Type\MoneyType;
+
+// ...
+
+$zboziFeedProductFormBuilder->add('cpc', MultidomainType::class, [
+    'label' => $this->translator->trans('Maximum price per click'),
+    'entry_type' => MoneyType::class,
+    'required' => false,
+    'entry_options' => [
```

Comment:
I think it would be nice to have here one of `{min/max}Message` to ilustrate where the option belongs

Reply:
This is an example straight from the actual code. It wouldn't make much sense to specify only one message (as both `min` and `max` are used). I could use an entirely different example to show how to use it (eg. specify only `min` and `minValue`) but I don't see much value.

In my opinion, people are used to the `Range` constraint and the usage is pretty straight-forward.